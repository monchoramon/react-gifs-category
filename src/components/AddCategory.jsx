import { useState } from 'react'
import PropType from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    // --- Enter y agregar a la lista
    const onSubmit = (evt) => {

        evt.preventDefault();

        let trimInputValue = inputValue.trim();

        if (trimInputValue.length > 1) {
            setInputValue('');
            onNewCategory(trimInputValue);
        } else {
            alert("La categoría debe de tener una longitud mínima de 2 caracteres.");
        }

    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder='Buscar gifs'
                value={inputValue}
                onChange={onInputChange} />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropType.func.isRequired
}