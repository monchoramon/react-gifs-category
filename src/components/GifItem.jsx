import PropTypes from 'prop-types';

export const GifItem = ({ title, url, id }) => {
    return (
        <div className="card">
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string
}

GifItem.defaultProps = {
    title: "Back To The Future GIF",
    url: "https://media2.giphy.com/media/xsF1FSDbjguis/giphy.gif?cid=505daff6au6re92h99hyrpdtttcdqausyji2at27n4ot9xmk&ep=v1_gifs_search&rid=giphy.gif&ct=g",
}