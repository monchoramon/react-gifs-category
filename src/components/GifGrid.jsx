import PropTypes from "prop-types";
import { GifItem } from '../components/GifItem';
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

    const { images, isLoading, findImages } = useFetchGifs(category);

    return (
        <>
            <hr />
            <h3>Categoría:</h3>
            <h4 className='categoryName'>{">" + category}</h4>
            <br />
            {
                isLoading && <b>Cargando...</b>
            }
            <div className="card-grid">
                {
                    images.map((image, index) =>
                    (
                        <GifItem
                            key={image.id}
                            index={index}
                            {...image} />
                    ))
                }
                {
                    !findImages &&
                    <h4 className='alert-warning'>
                        <p>
                            <b>¡No hay imágenes para mostrar!</b>
                        </p>
                    </h4>
                }
            </div>
        </>
    )

}

GifGrid.prototypes = {
    category: PropTypes.string.isRequired
}
