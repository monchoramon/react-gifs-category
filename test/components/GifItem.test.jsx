import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en el componente <GifItem />', () => {

    const title = 'Back To The Future GIF';
    const url = 'https://media2.giphy.com/media/xsF1FSDbjguis/giphy.gif?cid=505daff6au6re92h99hyrpdtttcdqausyji2at27n4ot9xmk&ep=v1_gifs_search&rid=giphy.gif&ct=g';

    test('Debe de hacer match con el snapshot ', () => {

        const { container } = render(
            <GifItem
                title={title}
                url={url} />
        );

        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url} />);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('Debe de mostrar el tÃ­tulo en el componente', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

    });

});