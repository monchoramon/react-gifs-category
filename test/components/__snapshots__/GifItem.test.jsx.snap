// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem /> Debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Back To The Future GIF
    </p>
    <img
      alt="Back To The Future GIF"
      src="https://media2.giphy.com/media/xsF1FSDbjguis/giphy.gif?cid=505daff6au6re92h99hyrpdtttcdqausyji2at27n4ot9xmk&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
  </div>
</div>
`;
